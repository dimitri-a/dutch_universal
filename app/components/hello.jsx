import React from 'react';

export default class Hello extends React.Component {

    constructor(props) {
        super();
        this.props = props;
        //todo remove
        console.log('testing props', props);
    }

    render() {
        let greet ;

        if (this.props) {
            greet= this.props.timeofday ==='am' ? 'morning' : 'evening'
        }
        return <h1>Good {greet}</h1>
    }
}