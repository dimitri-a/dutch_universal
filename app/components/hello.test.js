import React, { Component } from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import {Hello} from './hello';

test('if am then return goodmorning', () => {

    const wrapper = shallow(
        <Hello timeofday="am">
            <strong>Hello World!</strong>
        </Hello>
    );

    expect(toJson(wrapper)).toMatchSnapshot();

});