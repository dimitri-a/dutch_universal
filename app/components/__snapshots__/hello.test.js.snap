// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`if am then return goodmorning 1`] = `
<h1>
  Good 
  morning
   , 
  <h3>
    jojo
  </h3>
</h1>
`;

exports[`if pm then return good evening 1`] = `
<h1>
  Good 
  evening
   , 
  <h3>
    jojo
  </h3>
</h1>
`;
